### DISCUSSION TITLE
smallrye config question

### DISCUSSION THREAD

**Discussion by guykatz** [quarkusio/quarkus#28741] [Community]
Hi; following insights episode 103 the debezium demo code needs to read properties from the application.properties with a specific prefix into a properties object.
i would assume that we would have something like
`
    @ConfigProperties(prefix="dbzm")
    public Properties<String, String> inj;
`
which would inject to the properties object all props from the file starting wiht the specified prefix (minus the porefix)
however, they actually coded the conversion by reading all properties, substringing them and building the properties "from scrach"
i tried to find the way to achieve this with simple injection and didnt find the way.
is this missing in smallrye/mp config?
again assuming we have application.properties as such:
`
dbzm.database.hostname=10.110.116.22
dbzm.database.port=5432
dbzm.database.user=nms
`
I would like to do something like:
`
    @ConfigProperties(prefix="dbzm")
    public Properties<String, String> inj;
`
and use this properties object where needed.
looks straight forward to me. am I missing something?
thanks

**Comment by itatdcer** [comment]
You could try to define the prefix dbzm.database and, instead of a Properties, create a pojo with hostname, port and user properties.

**Reply by guykatz** [reply]
Tnx. This option is actually not good enough for cases like these...
The whole issue is that the number and names of the properties are not known in advance and should be read via the prefix.
Debezium have many connectors each with their own properties...

**Comment by radcortez** [comment]
You can do something like this:

```java
@ConfigMapping(prefix = "dbzm")
public interface Dbzm {
    @WithParentName
    Map<String, String> properties();
}
```

```properties
dbzm."database.hostname"=10.110.116.22
dbzm."database.port"=5432
dbzm."database.user"=nms
```

Note that the key segment must be quoted, so the entire segment is handled as a key entry in the Map.

**Reply by guykatz** [reply]
@radcortez  thanks. looks interesting... i will check this option out.
I still also posted a post in the MP group. I think having
@configProperties (pref="")
Properties <> props;
should be a part of MP if the implementation is strightforward...
thanks!

**Reply by radcortez** [reply]
I've also replied to your email in the MP group. The MP Config spec has no support for Map like constructs. The only option is to build it manually by iterating `Config.getPropertyNames()`, filtering the property name and retrieving the value and placing it into a `Properties` or `Map`.
